homework2/task01/findInFile/findInFile.py

```diff
@@ -0,0 +1,77 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+import re
+import string
+from collections import Counter
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    list_of_longest = set()
```

Comment:
I'd try another approach. What if you just read input line-by-line, extract all words from each line, and then fill the `dict` with `words consisting from ... unique symbols`, where key is number of symbols, and value is a list of the words.

Having filled the `dict`, you can easily extract desired number of words.

Reply:
I've created new function and stuck on this
```suggestion
    def get_longest_diverse_words_dict_temp(file_path: str) -> List[str]:
    # read all words from file to list
    words = {}
    with open(file_path, encoding="unicode_escape", errors="backslashreplace") as fi:
        for line in fi:
            inner_list = re.findall(r"[\w-]+", line)
            for word in inner_list:
                words.setdefault(len(word), set()).add(word)
    return words
```
First, I have an issue with words that start on one line and end on the next. It was easy to fix in list (if word is ending with symbol "-" just merge it with the following word) and now I have some suggestion like pre-read this file and replace all '-\n' by regular expression and some others, but they all look not good for me.
Second, I'm not sure how we can extract required number of words from dictionary (10 by example). Should I sorted it by keys and fill some new list with values from dictionary?